packages/openapi-generator/src/generator.ts

```diff
@@ -30,30 +32,34 @@ const logger = createLogger('openapi-generator');
  * Generates files using the OpenApi Generator CLI and wraps the resulting API in an SDK compatible API.
  * @param options Options to configure generation.
  */
-export async function generate(options: GeneratorOptions): Promise<void> {
+export async function generate(options: GeneratorOptions): Promise<void[]> {
   options.serviceMapping =
     options.serviceMapping ||
     resolve(options.input.toString(), 'service-mapping.json');
 
   if (options.clearOutputDir) {
-    await rmdir(options.outputDir, { recursive: true });
+    if (existsSync(options.outputDir)) {
+      await rmdir(options.outputDir, { recursive: true });
+    }
   }
 
   const vdmMapping = readServiceMapping(options);
   const uniqueNameGenerator = new UniqueNameGenerator('-');
   const inputFilePaths = await getInputFilePaths(options.input);
-  inputFilePaths.forEach(async inputFilePath => {
+
+  const promises = inputFilePaths.map(async inputFilePath => {
     const uniqueServiceName = uniqueNameGenerator.generateAndSaveUniqueName(
       parseServiceName(inputFilePath)
     );
 
-    await generateFromFile(
+    return generateFromFile(
```

Comment:
I guess it is intended to `await` as the original implementation, since running in parallel will result in race condition in terms of the unique service name check

Reply:
I saw in the test, that the generation that the await did not have the desired effect. This was just a test. I made a Promise.all which really resolves once all are finished.